docs/fatjar-regressions/fatjar-regressions-v0.36.1-SNAPSHOT.md

```diff
@@ -25,48 +25,96 @@ The `msmarco-v2.1-doc` prebuilt index is 63 GB uncompressed.
 The `msmarco-v2.1-doc-segmented` prebuilt index is 84 GB uncompressed.
 Both indexes will be downloaded automatically.
 
-Here are the instructions for reproducing runs on the MS MARCO V2.1 document corpus with prebuilt indexes (adjust number of threads based on available resources):
+For the TREC 2024 RAG track, we have thus far only implemented BM25 baselines on the MS MARCO V2.1 document corpus (both the doc and doc segmented variants).
+Current results are based existing qrels that have been "projected" over from MS MARCO V2.0 passage judgments.
+The table below reports effectiveness (dev in terms of RR@10, DL21-DL23, RAGgy in terms of nDCG@10):
+
+|                                                                            |    dev |   dev2 |   DL21 |   DL22 |   DL23 |  RAGgy |
+|:---------------------------------------------------------------------------|-------:|-------:|-------:|-------:|-------:|-------:|
+| BM25 doc (<i>k<sub><small>1</small></sub></i>=0.9, <i>b</i>=0.4)           | 0.1654 | 0.1732 | 0.5183 | 0.2991 | 0.2914 | 0.3631 |
+| BM25 doc-segmented (<i>k<sub><small>1</small></sub></i>=0.9, <i>b</i>=0.4) | 0.1973 | 0.2000 | 0.5778 | 0.3576 | 0.3356 | 0.4227 |
+
+The follow command will reproduce the above experiments:
 
 ```bash
-TOPICS=(msmarco-v2-doc-dev msmarco-v2-doc-dev2 trec2021-dl trec2022-dl trec2023-dl rag24-raggy-dev); for t in "${TOPICS[@]}"
-do
-    java -cp $ANSERINI_JAR io.anserini.search.SearchCollection -index msmarco-v2.1-doc -topics $t -output $OUTPUT_DIR/run.msmarco-v2.1.doc.${t}.txt -threads 16 -bm25
-done
+java -cp $ANSERINI_JAR io.anserini.reproduce.RunMsMarco -collection msmarco-v2.1
```

Comment:
Getting `Run failed!` and consequently `Evaluation command failed` for some retrievals. For instance:
```
# Running condition "doc-segmented": BM25 segmented doc (k1=0.9, b=0.4)           
                                                                                             
  - topic_key: msmarco-v2-doc.dev                                                            
                                                                                             
    Running retrieval command: java -cp /u9/s42chen/coding/anserini/target/anserini-0.36.1-SN
APSHOT-fatjar.jar io.anserini.search.SearchCollection -threads 16 -index msmarco-v2.1-doc-seg
mented -topics msmarco-v2-doc.dev -output runs/run.msmarco-v2.1.doc-segmented.msmarco-v2-doc.
dev.txt -hits 10000 -bm25 -selectMaxPassage -selectMaxPassage.delimiter \# -selectMaxPassage.
hits 1000              
    Run failed!                               

Evaluation command failed for metric: MRR@10    
```
However, running the command separately gives:
```
(java21_env) s42chen@ubuntu2204-002:~/coding/anserini$ java -cp /u9/s42chen/coding/anserini/target/anserini-0.36.1-SNAPSHOT-fatjar.jar io.anserini.search.SearchCollection -threads 16 -index msmarco-v2.1-doc-segmented -topics msmarco-v2-doc.dev -output runs/run.msmarco-v2.1.doc-segmented.msmarco-v2-doc.dev.txt -hits 10000 -bm25 -selectMaxPassage -selectMaxPassage.delimiter \# -selectMaxPassage.hits 1000   
Index file already exists! Skip downloading.
Index folder already exists!
2024-05-11 19:24:38,345 INFO  [main] search.SearchCollection (SearchCollection.java:1008) - ============ Initializing Searcher ============
2024-05-11 19:24:38,388 INFO  [main] search.SearchCollection (SearchCollection.java:1009) - Index: /u9/s42chen/.cache/pyserini/indexes/lucene-inverted.msmarco-v2.1-doc-segmented.20240418.4f9675.6ec4cd595c9fe1ad91b43eabb39a637c
May 11, 2024 7:24:38 PM org.apache.lucene.store.MemorySegmentIndexInputProvider <init>
INFO: Using MemorySegmentIndexInput with Java 21; to disable start with -Dorg.apache.lucene.store.MMapDirectory.enableMemorySegments=false
2024-05-11 19:24:39,828 INFO  [main] search.SearchCollection (SearchCollection.java:1012) - Threads: 16
2024-05-11 19:24:39,829 INFO  [main] search.SearchCollection (SearchCollection.java:1013) - Fields: []
2024-05-11 19:24:39,830 INFO  [main] search.SearchCollection (SearchCollection.java:1027) - MaxPassage: true
2024-05-11 19:24:39,831 INFO  [main] search.SearchCollection (SearchCollection.java:1029) - MaxPassage delimiter: #
2024-05-11 19:24:39,832 INFO  [main] search.SearchCollection (SearchCollection.java:1030) - MaxPassage hits: 1000
2024-05-11 19:24:39,832 INFO  [main] search.SearchCollection (SearchCollection.java:1032) - Hits: 10000
2024-05-11 19:24:39,833 INFO  [main] search.SearchCollection (SearchCollection.java:1045) - Collection class: null
2024-05-11 19:24:39,857 INFO  [main] search.SearchCollection (SearchCollection.java:1332) - Using DefaultEnglishAnalyzer
2024-05-11 19:24:39,858 INFO  [main] search.SearchCollection (SearchCollection.java:1333) - Stemmer: porter
2024-05-11 19:24:39,859 INFO  [main] search.SearchCollection (SearchCollection.java:1334) - Keep stopwords? false
2024-05-11 19:24:39,861 INFO  [main] search.SearchCollection (SearchCollection.java:1335) - Stopwords file: null
2024-05-11 19:24:40,206 INFO  [main] search.SearchCollection (SearchCollection.java:1345) - ============ Launching Search Threads ============
2024-05-11 19:24:40,207 INFO  [main] search.SearchCollection (SearchCollection.java:1346) - runtag: Anserini
2024-05-11 19:35:36,560 INFO  [pool-3-thread-14] search.SearchCollection$SearcherThread (SearchCollection.java:904) - ranker: bm25(k1=0.9,b=0.4), reranker: default: 100 queries processed
CPU time limit exceeded
```
Which seems to be just the student server killing it. Should I run it again on WSL tomorrow?

Reply:
Nah, I think we're good!